python/src/aiconfig/util/config_utils.py

```diff
@@ -10,10 +11,28 @@
     from ..schema import AIConfig
 
 
-def get_api_key_from_environment(api_key_name: str):
-    if api_key_name not in os.environ:
-        raise Exception("Missing API key '{}' in environment".format(api_key_name))
+def maybe_get_api_key_from_environment(
+    api_key_name: str,
+    required: bool = True) -> Union[str, None]:
+    """Get the API key if it exists, return None or error if it doesn't
+
+    Args:
+        api_key_name (str): The keyname that we're trying to import from env variable
+        required (bool, optional): If this is true, we raise an error if the 
+            key is not found
+
+    Returns:
+        Union[str, None]: the value of the key. If `required` is false, this can be None
+    """
+    dotenv.load_dotenv()
+    if required:
+        _get_api_key_from_environment(api_key_name)
+    return os.getenv(api_key_name)
 
+
+def _get_api_key_from_environment(api_key_name: str) -> str:
+    if api_key_name not in os.environ:
+        raise KeyError(f"Missing API key '{api_key_name}' in environment")
```

Comment:
Don't raise and return Optional. You can just do this:

```
def get_api_key_from_environment(api_key_name: str) -> str | None:
    try:
         return os.environ[api_key_name]
    except Exception:
         # you want to do the logging here because you're about to lose the message forever by returning `None`.
         LOGGER.error(...)
          return None
```